test/replication-luatest/gh_9918_synchro_queue_additional_info_test.lua

```diff
@@ -0,0 +1,102 @@
+local fiber = require('fiber')
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_each(function(cg)
+    cg.replica_set = replica_set:new{}
+    local box_cfg = {
+        replication = {
+            server.build_listen_uri('server1', cg.replica_set.id),
+            server.build_listen_uri('server2', cg.replica_set.id),
+        },
+        replication_timeout = 0.1,
+    }
+    cg.leader = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = box_cfg,
+    }
+    cg.replica = cg.replica_set:build_and_add_server{
+        alias = 'server2',
+        box_cfg = box_cfg,
+    }
+    cg.replica_set:start()
+    cg.leader:exec(function()
+        box.ctl.promote()
+        local s = box.schema.space.create('s', {is_sync = true})
+        s:create_index('pk')
+    end)
+    cg.replica_set:wait_for_fullmesh()
+    cg.leader:wait_for_downstream_to(cg.replica)
+end)
+
+g.after_each(function(cg)
+    cg.replica_set:drop()
+end)
+
+-- Test that new `age` field of `box.info.synchro.queue` works correctly.
+g.test_age_field = function(cg)
+    -- The synchronous queue is originally empty.
+    cg.leader:exec(function()
+        t.assert_equals(box.info.synchro.queue.age, 0)
+    end)
+    cg.replica:exec(function()
+        t.assert_equals(box.info.synchro.queue.age, 0)
+    end)
+
+    -- Ensure that the synchronous transactions stay in the queue for a while.
+    cg.leader:update_box_cfg{replication_synchro_quorum = 3,
+                             replication_synchro_timeout = 60}
+    -- Add the first entry to the synchronous queue.
+    local fid1, oldest_entry_ts = cg.leader:exec(function()
+        local fiber = require('fiber')
+
+        local f = fiber.new(function() box.space.s:replace{0} end)
+        f:set_joinable(true)
+        return f:id(), fiber.clock()
+    end)
+    local wait_time = 1
+    fiber.sleep(wait_time)
```

Comment:
One second is a very long time for such a trivial test. Please, make it something like 0.1. And Nikita is right about being hardly possible to check the time for comparison. Unfortunately, the time is completely unpredictable. You can only test reliably very basic things - that the age is > the wait time, and it is, say, < 2 minutes.

Reply:
Fixed. Testing that the age > the wait time seems to also be unreliable, so I just test it is > 0.

Reply:
Hm. But how is that possible? If you waited for N seconds, you couldn't have committed it faster than that. You would need to sleep less than this time. Could you find the reason please? It could mean a bug.

Reply:
We can test this on the leader (changed the tests from `0` to `wait_time`, where applicable), but we cannot test this on the replica — since the limbo entry arrives on the replica with some delay, and the timekeeping there starts later.